sys/net/network_layer/sixlowpan/icmp.c

```diff
@@ -1221,17 +1222,17 @@ void icmpv6_send_neighbor_adv(ipv6_addr_t *src, ipv6_addr_t *dst, ipv6_addr_t *t
 
     packet_length = IPV6_HDR_LEN + ICMPV6_HDR_LEN + NBR_ADV_LEN;
 
-    if (sllao == OPT_SLLAO) {
+    if (sllao == OPT_TLLAO) {
         /* set sllao option */
         opt_stllao_buf = get_opt_stllao_buf(ipv6_ext_hdr_len, icmpv6_opt_hdr_len);
 
         if (net_if_get_src_address_mode(if_id) == NET_IF_TRANS_ADDR_M_LONG) {
-            icmpv6_ndp_set_sllao(opt_stllao_buf, if_id, NDP_OPT_SLLAO_TYPE, 2);
+            icmpv6_ndp_set_sllao(opt_stllao_buf, if_id, NDP_OPT_TLLAO_TYPE, 2);
             icmpv6_opt_hdr_len += OPT_STLLAO_MAX_LEN;
             packet_length += OPT_STLLAO_MAX_LEN;
         }
         else {
-            icmpv6_ndp_set_sllao(opt_stllao_buf, if_id, NDP_OPT_SLLAO_TYPE, 1);
+            icmpv6_ndp_set_sllao(opt_stllao_buf, if_id, NDP_OPT_TLLAO_TYPE, 1);
```

Comment:
I can not confirm it. It is used in various places. I do not really have an overview of 1872 lines as icmp newbie.